services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedTenantService.java

```diff
@@ -0,0 +1,572 @@
+        this.mongoClient = Objects.requireNonNull(this.mongoDbCallExecutor.getMongoClient());
+    }
+
+    public MongoDbBasedTenantsConfigProperties getConfig() {
+        return config;
+    }
+
+    /**
+     * Sets the configuration properties for this service.
+     *
+     * @param config The configuration properties.
+     * @throws NullPointerException if the configuration is {@code null}.
+     */
+    @Autowired
+    public void setConfig(final MongoDbBasedTenantsConfigProperties config) {
+        this.config = Objects.requireNonNull(config);
+    }
+
+    @Override
+    public void start(final Promise<Void> startPromise) {
+        mongoDbCallExecutor.createCollectionIndex(config.getCollectionName(),
+                new JsonObject().put(RegistrationConstants.FIELD_PAYLOAD_TENANT_ID,
+                        1),
+                new IndexOptions().unique(true), INDEX_CREATION_MAX_RETRIES)
+                .map(success -> {
+                    startPromise.complete();
+                    return null;
+                })
+                .onFailure(error -> {
+                    log.error("Index creation failed", error);
+                    startPromise.fail(error);
+                });
+    }
+
+    @Override
+    public void stop(final Promise<Void> stopPromise) {
+        this.mongoClient.close();
+        stopPromise.complete();
+    }
+
+    /**
+     * Updates configuration information of a tenant.
+     *
+     * @param tenantId The identifier of the tenant.
+     * @param tenantObj The updated configuration information for the tenant (may be {@code null}).
+     * @param resourceVersion The identifier of the resource version to update.
+     * @param span The active OpenTracing span for this operation. It is not to be closed in this method! An
+     *            implementation should log (error) events on this span and it may set tags and use this span as the
+     *            parent for any spans created in this method.
+     * @return A future indicating the outcome of the operation.
+     *         The <em>status</em> will be
+     *         <ul>
+     *         <li><em>204 No Content</em> if the tenant has been removed successfully.</li>
+     *         <li><em>403 forbidden</em> if tenant modification is forbidden due to configuration settings</li>
+     *         <li><em>404 Not Found</em> if no tenant with the given identifier and version exists.</li>
+     *         <li><em>409 Conflict</em> if provided trusted certificate authority is already present in another
+     *         tenant.</li>
+     *         <li><em>412 Precondition Failed</em> if the version exists but mismatch the current tenants version.</li>
+     *         </ul>
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     * @see <a href="https://www.eclipse.org/hono/docs/api/management/#/tenants/updateTenant"> Device Registry
+     *      Management API - Update Tenant</a>
+     */
+    @Override
```

Comment:
In general for the overridden methods, the _JavaDoc_ of the original methods could be used and no need for another _JavaDoc_ in the overridden methods. The additional information could be added using {@inheritDoc}. For example, please refer to `VertxBasedCoapAdapter.getTypeName()`.

In this case, I see that you have added status codes which were missing in the `TenantManagementService` but specified in the swagger api. I would suggest that you go ahead and update the javdoc of the `TenantManagementService` and remove the _JavaDoc_  here. 

Also please check the _JavaDocs_ of the other overridden methods.

Reply:
done